.pre-commit.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env sh
+# pre-commit.sh
+
+STASH_NAME="pre-commit-$(date +%s)"
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m'
+BOLD='\033[1m'
+
+stash=0
+# Check to make sure commit isn't emtpy, exit with status 1 if it is
+if git diff-index --quiet HEAD --; then
+    echo "${RED}You've tried to commit an empty commit${NC}"
+    echo "\tMake sure to add your changes with 'git add'"
+    exit 1
+else
+    # Stash all changes in the working directory so we test only commit files
+    if git stash save -u -k -q $STASH_NAME; then
+        echo "${YELLOW}Stashed changes as:${NC} ${STASH_NAME}\n\n"
+        stash=1
+    fi
+fi
+
```

Comment:
Good catch!  
I found a [blog post](http://gmurphey.com/2013/02/02/ignoring-git-hooks-when-rebasing.html) with a solution, will test it when I get home.